gc/stats/CollectionStatistics.hpp

```diff
@@ -63,10 +69,40 @@ class MM_CollectionStatistics : public MM_Base
 		,_startTime(0)
 		,_endTime(0)
 		,_stallTime(0)
+		,_pauseStart(0)
+		,_pauseEnd(0)
+		,_pauseLength(0)
+		,_pauseTotal(0)
+		,_pauseLongest(0)
 		,_startProcessTimes()
 		,_endProcessTimes()
 		,_cpuUtilStats()
 	{};
+
+	void
+	recordPauseStart(uint64_t startTime)
+	{
+		_pauseStart = startTime;
+	}
+
+	void
+	processPauseEnd(uint64_t endTime)
+	{
+		_pauseEnd = endTime;
+		_pauseLength = _pauseEnd - _pauseStart;
```

Comment:
I think this calculation should be protected from backwards wall clock correction, we don't want `_pauseTotal` go backwards:
```
    if (_pauseEnd > _pauseStar) {
        _pauseLength = _pauseEnd - _pauseStart;
    } else {
        _pauseLength = 0;
    }
```